gdc_client/upload/client.py

```diff
@@ -51,12 +51,8 @@ def __init__(self):
 log = logging.getLogger("upload-client")
 
 
-def upload_multipart_wrapper(args):
-    return upload_multipart(*args)
-
-
 class Stream(object):
-    def __init__(self, file, pbar, filesize):
+    def __init__(self, file, pbar: tqdm, filesize: int):
```

Comment:
I think this class is one of the problems with upload progress bar. `Request` demands a file-like object for streaming upload, see [docs](https://2.python-requests.org/en/master/user/advanced/#streaming-uploads) this class simply implements `read` with a parameter. I tried to verify that this is all `requests` requires to figure out if an object is file-like or not and it led me to this [implementation](https://github.com/psf/requests/blob/1ca1c52e698b13d3d5cc0755a6450306d880b933/requests/models.py#L471-L494)  which looks like it expects an `__iter__` and `tell` method instead of a `read` with a parameter. If this is correct, the way the class is implemented is why upload jumps from zero to completion

Reply:
I don't think we use this class for multi-part uploads. not sure if the progress bar was an issue for regular non-multi-part uploads. I'll take a look

Reply:
turns out the `__getattr__` takes care of everything. the only concern I had is an implicit `__iter__` definition, but it doesn't look like `requests` is using it that way. I did manual pdb'ing to make sure. so we're good with the current implementation